doc/nrf/libraries/zigbee/lib_zigbee_signal_handler.rst

```diff
@@ -0,0 +1,303 @@
+    The procedure will be restarted only if the device does not join and the procedure is not running.
+
+The Zigbee rejoin procedure retries to join a network with each try after a specified amount of time: ``2^n`` seconds, where ``n`` is the number of retries.
+
+The period is limited to 15 minutes if the result is higher than that.
+
+* When :cpp:func:`start_network_rejoin()` is called, the rejoin procedure is started, and depending on the device role:
+
+  * For the end device, the application alarm is scheduled with ``stop_network_rejoin(ZB_TRUE)``, to be called after the amount of time specified in ``ZB_DEV_REJOIN_TIMEOUT_MS``.
+    Once called, the alarm stops the rejoin.
+
+* When ``stop_network_rejoin(was_scheduled)`` is called, the network rejoin is canceled and the alarms scheduled by :cpp:func:`start_network_rejoin()` are canceled.
+
+  * Additionally for the end device, if :cpp:func:`stop_network_rejoin()` is called with ``was_scheduled`` set to ``ZB_TRUE``, :cpp:func:`user_input_indicate()` can restart the rejoin procedure.
+
+* For end devices only, :cpp:func:`user_input_indicate()` restarts the rejoin procedure if the device did not join the network and is not trying to join a network.
+  It is safe to call this function from an interrupt and to call it multiple times.
+
+
+.. warning::
+    The Zigbee network rejoin procedure is managed from multiple signals in :cpp:func:`default_signal_handler()`.
+    If the application controls the network joining, rejoining, or leaving, each signal in which the Zigbee network rejoin procedure is managed should be handled in the application.
+    In this case, :cpp:func:`user_input_indicate()` must not be called.
+
+.. _zarco_sleep:
+
+Zigbee stack sleep routines
+***************************
+
+The Zigbee stack provides the following mechanisms to reduce power consumption:
+
+* On Zigbee end devices, routers and coordinators:
+
+  * The stack calls the Power Management API (through :cpp:func:`zb_osif_wait_for_event`) if the Zigbee scheduler queue is empty.
+
+* On Zigbee sleepy end devices:
+
+  * If there are no immediate callbacks scheduled through the Zigbee scheduler, the stack calls the Power Management API (through :cpp:func:`zb_osif_wait_for_event`).
+  * If there are no delayed callbacks (alarms) scheduled for the next ``sleep_threshold`` period, the stack switches off all peripheral devices related to the Zigbee stack before calling the Power Management API.
+
+On Zigbee sleepy end devices, the ``sleep_threshold`` period lasts approximately 15 ms by default and can be modified by using the ``zb_sleep_set_threshold`` API.
+
+If the device implements a Zigbee sleepy end device, the ``ZB_COMMON_SIGNAL_CAN_SLEEP`` signal is generated whenever the device can be put into the deep sleep mode.
```

Comment:
To be improved:
 - The reasoning behind the sleep signal is no longer focused on energy consumption. We implement it for all device types, so the ZBOSS task does use CPU time.
 - The sleep threshold applies to all device types.
 - ZBOSS OSIF does not call Power Management API.
 - ZBOSS OSIF does not manage shared resources. The Radio and timer state are managed by ZBOSS core.